### PR TITLE
Add config to run conmon under a custom cgroup slice

### LINKED ISSUE #2248: Support specifying custom cgroups for conmon
Today, we nest conmon under the pod cgroup. However, some installations may not desire that and may want to put all the conmon processes under a different cgroup; for e.g. system.slice or some dedicated conmon.slice that isn't charged to the pod.

We can add a `conmon_cgroup` setting to support that. If it is set to `pod` or is empty, then we nest it under the pod, else we use the slice specified there.

### PR DESCRIPTION
**Pull request by mrunalp**
Fixes #2248 

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by mrunalp** [comment]
@sboeuf @chavafg kata jobs have been failing.

File: lib/config.go [hunk 1]
```diff
@@ -383,6 +386,7 @@ func DefaultConfig() (*Config, error) {
 			ConmonEnv: []string{
 				"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
 			},
+			ConmonCgroup:             "pod",
```
[the remark this answers is not in the document]
Comment (mrunalp) [reply lib/config.go]: The config code is messed up and spread in such a way that it leads to a circular import. I will fix that up in a follow-on PR to refactor it.

**Comment by mrunalp** [comment]
@giuseppe Added tests. This is ready now.

**Comment by giuseppe** [comment]
LGTM

**Comment by mrunalp** [comment]
@umohnani8 ptal

**Comment by haircommander** [comment]
LGTM

**Comment by haircommander** [comment]
/cherrypick release-1.14
maybe? :D

**Comment by haircommander** [comment]
:cry:

**Pull request merged**